ord_schema/proto/reaction.proto

```diff
@@ -861,26 +860,136 @@ message ReactionProduct {
+    // the m/z ratio(s) should be defined in the details.
+    COUNTS = 7;
+    // Intensity, e.g., of a TLC spot, ELSD, or GC FID.
+    INTENSITY = 8;
   }
-  SelectivityType type = 1;
-  string details = 2;
-  optional float value = 3;
-  // TODO(kearnes): What does precision mean in this context?
-  optional float precision = 4;
+  MeasurementType type = 2;
+  // Additional information about the measurement and, in particular, how it
+  // was calculated if that is ambiguous given the analysis type.
+  string details = 3;
+  // Whether an internal standard was used for this measurement to normalize
+  // it, e.g., for quantification of yield or relative peak area. If this
+  // field is true, the value field should contain the normalized quantity.
+  optional bool uses_internal_standard = 4;
+  // Whether an authentic sample of a material was used for callibrating
+  // an analytical technique, e.g., for product identification by LCMS.
+  // "Authentic" refers to a sample of a compound whose structure has been
+  // proven by the accepted means (1H NMR, 13C NMR, IR, HRMS), and is now
+  // being used as a comparison for other analytical methods.
+  optional bool uses_authentic_standard = 5;
+  // Details about the authentic standard used, optionally including
+  // information about its precise salt form and source/vendor.
+  Compound authentic_standard = 6;
+  // The value for this measurement.
+  oneof value {
+    // A percentage value and its precision, as should be used for yields,
+    // purities, and some selectivities like EE. Note that a 50% yield
+    // corresponds to a percentage.value of 50 and not 0.5.
+    Percentage percentage = 7;
+    // A float value and its precision, as should be used for areas, counts,
+    // intensities, and some selectivities.
+    FloatValue float_value = 8;
+    // A string value, as should be used for measurements that cannot be
+    // captured as floats or percentages. For example, complex selectivities
+    // such as "2:1:1". Note that in the case of a 2:1:1 selectivity, however,
+    // it is preferable to define the 3 distinct products specifically and
+    // record the 3 product-specific measurements (e.g., peak areas) that
+    // were used to derive the "2:1:1" summary statistic.
+    string string_value = 9;
+
+  }
+  // A time to be used only when specifying the retention time at which a peak
+  // area or intensity was calculated, e.g., as in LC, LCMS, GC, or GCMS.
+  Time retention_time = 10;
+  // More details about how a particular performance metric (e.g., counts)
+  // was calculated from an analysis using mass spec. This includes
+  // distinguishing total ion current (TIC) counts from extracted ion
+  // chromatogram (EIC) counts and their corresponding mass ranges or masses.
+  message MassSpecType {
+    enum MassSpecTypeEnum {
+      UNSPECIFIED = 0;
+      CUSTOM = 1;
+      TIC = 2;
+      TIC_POS = 3;
```

Comment:
nit: I think it's fine to use accepted acronyms, but it's good practice to avoid _abbreviations_ (see https://google.github.io/styleguide/pyguide.html#316-naming). So I'd suggest e.g. `TIC_POSITIVE`.